packages/amplify-cli/src/utils/verify-expected-env-params.ts

```diff
@@ -27,7 +28,16 @@ export const verifyExpectedEnvParams = async (context: $TSContext, category?: st
   });
 
   if (context?.exeInfo?.inputParams?.yes || context?.exeInfo?.inputParams?.headless) {
-    await envParamManager.verifyExpectedEnvParameters(parametersToCheck);
+    let appId: string | undefined = undefined;
+    try {
+      appId = resolveAppId(context);
+    } catch {
+      // If AppId can't be resolved, this only affects the error message of verifyExpectedEnvParameters in the case that parameters are
+      // actually missing. So we let appId be undefined here and verifyExpectedEnvParameters will print a different error message based
+      // on the information available to it
+    }
```

Comment:
does `resolveAppId` try inputParams too ? I.e. why don't we fallback back/check inputs like in other places ?

Reply:
yes resolveAppId checks input params internally. However, because it's in the cli package it's not reuseable in other places